src/com/sap/piper/ConfigurationHelper.groovy

```diff
@@ -100,6 +100,15 @@ class ConfigurationHelper implements Serializable {
         return this
     }
 
+    ConfigurationHelper addIfNull(key, value){
+        if (config[key] instanceof Boolean) {
```

Comment:
What is the reason for handling Booleans in a special way. From the naming I would expect something checking a collection/map if a value is already present and only to add if there is no such entry.

Reply:
I still do not understant the code here.

First we check if there is a boolean value associated with the key. If so we do nothing.
Later down the road we do nothing if there is _anything_ associated with the key. In this case we do nothing also.
Only if there is nothing associated with the key we set our value.

If there is a boolean associated with the key there is _something_ ( --> ! _anything_)  associdated with key.

IMO this code is fully equivalent to having the same code without the first if:

```
if(config[key] == null)
  config[key] = value
return this
```

Convince me from the opposite.

We should either provide a good reason for having the code as it is proposed here or we should not merge this.

Next question would be: what should happen if the null value was intentionally associated with `key`.  Does `addIfNull` mean the key is null (not present) or does it mean there can be the key, but with the null value. Not sure if that is relevant here and how we should deal with it.

Reply:
@marcusholl has been changed accordingly